userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -835,13 +833,29 @@ static int32_t load_bpf_file(
 	return res;
 }
 
-static void *perf_event_mmap(struct bpf_engine *handle, int fd, uint32_t *size)
+static void *perf_event_mmap(struct bpf_engine *handle, int fd, uint32_t *size, uint64_t buf_num_pages)
 {
+   /* Each data page is mapped twice to allow "virtual"
+	* continuous read of samples wrapping around the end of ring
+	* buffer area:
+	* 
+	* ------------------------------------------------------
+	* | meta pages |  real data pages  |  same data pages  |
+	* ------------------------------------------------------
+	* |            | 1 2 3 4 5 6 7 8 9 | 1 2 3 4 5 6 7 8 9 |
+	* ------------------------------------------------------
+	* |            | TA             DA | TA             DA |
+	* ------------------------------------------------------
+	*                               ^^^^^^^
+	*                                  |
+	* Here, no need to worry about special handling of wrapped-around
+	* data due to double-mapped data pages.
+	*/
 	int page_size = getpagesize();
-	int ring_size = page_size * BUF_SIZE_PAGES;
+	int ring_size = page_size * buf_num_pages;
 	int header_size = page_size;
 	int total_size = ring_size * 2 + header_size;
-	char buf[SCAP_LASTERR_SIZE];
+	char buf[SCAP_LASTERR_SIZE] = {0};
```

Comment:
Don't we need to call `set_per_cpu_buffer_dim` here?

Reply:
TBH this is not strictly necessary, but i've added it for completeness with a comment